sw/device/lib/testing/hmac_testutils.h

```diff
@@ -0,0 +1,85 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_TESTING_HMAC_TESTUTILS_H_
+#define OPENTITAN_SW_DEVICE_LIB_TESTING_HMAC_TESTUTILS_H_
+
+#include <assert.h>
+#include <stdint.h>
+
+#include "sw/device/lib/dif/dif_hmac.h"
+
+// Tests need to understand the byte order of the data in the string and
+// the digest values below, as they are laid out for the current processor.
+// RISC-V is little-endian, so the first of these `kHmacTransactionConfig`
+// values is the one we expect to be used, but we include the other for
+// completeness.
+#define HMAC_TESTUTILS_ASSERT_LITTLE_ENDIAN                \
+  static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__, \
+                "This test assumes the target platform is little endian.")
+/**
+ * FIFO empty 10uS timeout, which should be enough even for a 100kHz device.
+ *
+ * https://docs.opentitan.org/hw/ip/hmac/doc/
+ * Single HMAC block compression takes 80 cycles.
+ */
+#define HMAC_TESTUTILS_FIFO_EMPTY_TIMEOUT_USEC 10
```

Comment:
Is there a strong reason for these to be macros rather than enum constants?

Reply:
Nope, will change to enum.

Reply:
Done in #10995